File: src/main/java/com/phong/taskmanagement/service/DashboardService.java
package com.phong.taskmanagement.service;

import com.phong.taskmanagement.dto.response.DashboardResponse;
import com.phong.taskmanagement.dto.response.TaskMonthlyChartResponse;
import com.phong.taskmanagement.dto.response.TaskPriorityChartResponse;
import com.phong.taskmanagement.dto.response.TaskStatusChartResponse;

import java.util.List;

public interface DashboardService {
    DashboardResponse getDashboardStatistics();
    List<TaskStatusChartResponse> getTaskStatusChart();
    List<TaskPriorityChartResponse> getTaskPriorityChart();
    List<TaskMonthlyChartResponse> getTasksByMonthChart();
}

File: taskmanagement-ui/src/services/DashboardService.ts
import { apiClient } from '../api/axios';
import type { ApiResponse } from '../types/auth';
import type { 
  DashboardResponse, 
  TaskStatusChartResponse, 
  TaskPriorityChartResponse, 
  TaskMonthlyChartResponse 
} from '../types/dashboard';
import { API_ENDPOINTS } from '../utils/constants';

export const DashboardService = {
  getStatistics: async (): Promise<ApiResponse<DashboardResponse>> => {
    const response = await apiClient.get(API_ENDPOINTS.DASHBOARD_STATS);
    return response.data;
  },

  getTaskStatusChart: async (): Promise<ApiResponse<TaskStatusChartResponse[]>> => {
    const response = await apiClient.get(API_ENDPOINTS.DASHBOARD_TASK_STATUS);
    return response.data;
  },

  getTaskPriorityChart: async (): Promise<ApiResponse<TaskPriorityChartResponse[]>> => {
    const response = await apiClient.get(API_ENDPOINTS.DASHBOARD_TASK_PRIORITY);
    return response.data;
  },

  getTaskMonthlyChart: async (): Promise<ApiResponse<TaskMonthlyChartResponse[]>> => {
    const response = await apiClient.get(API_ENDPOINTS.DASHBOARD_TASKS_BY_MONTH);
    return response.data;
  },
};
